### PR TITLE
fix some optional values bugs

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by kshakir** [comment]
Not sure what this fixes, but code LGTM 👍

File: core/src/main/scala/cromwell/util/JsonFormatting/WdlValueJsonFormatter.scala [hunk 1]
```diff
@@ -18,6 +18,8 @@ object WdlValueJsonFormatter extends DefaultJsonProtocol {
       case m: WdlMap => new JsObject(m.value map {case(k,v) => k.valueString -> write(v)})
       case e: WdlExpression => JsString(e.toWdlString)
       case q: WdlPair => new JsObject(Map("left" -> write(q.left), "right" -> write(q.right)))
+      case WdlOptionalValue(_, Some(opt)) => write(opt)
+      case WdlOptionalValue(_, None) => JsString("null")
```
Comment (Horneth) [inline core/src/main/scala/cromwell/util/JsonFormatting/WdlValueJsonFormatter.scala]: No this one specifically is when the workflow outputs are printed to the console at the end of the workflow.

**Comment by Horneth** [comment]
It makes sure optional values are unboxed before being processed and be sent to the metadata.
Without this a defined optional array `[1, 2, 3]` would end up as a JsString `"[1, 2, 3]"` instead of JsArray in the metadata.

File: core/src/main/scala/cromwell/util/JsonFormatting/WdlValueJsonFormatter.scala (hunk 1 above)
Comment (mcovarr) [inline core/src/main/scala/cromwell/util/JsonFormatting/WdlValueJsonFormatter.scala]: @kshakir that's a very good point.  Aesthetically I prefer no k/v, but I bet an explicit k/"null" will probably reduce support questions.  So I'm not sure which to vote for...

**Comment by mcovarr** [comment]
:+1:

File: core/src/main/scala/cromwell/util/JsonFormatting/WdlValueJsonFormatter.scala (hunk 1 above)
Comment (geoffjentry) [inline core/src/main/scala/cromwell/util/JsonFormatting/WdlValueJsonFormatter.scala]: I could see support Qs either way. I think I'd prefer to see nothing

**Pull request merged**